src/r5py/r5/custom_cost_transport_network.py

```diff
@@ -0,0 +1,399 @@
+            If set to False and ANY edges aren't found during routing, routing will throw exception,
+            so in most cases the default True should be used.
+        """
+        # crash if custom costs are NOT supported in the used version of R5
+        # either use TransportNetwork (without custom costs) or change to correct version of R5
+        if not r5_supports_custom_costs():
+            raise RuntimeError(
+                """Custom costs are not supported in this version of R5.
+                Correct (Green Paths 2 patched) R5 version can be found from branch gp2 in
+                https://github.com/DigitalGeographyLab/r5. Or by using a release jar from address e.g.
+                https://github.com/DigitalGeographyLab/r5/releases/download/v7.1-gp2-1/r5-v7.1-gp2-2-gd8134d8-all.jar
+                """
+            )
+        # convert single items into lists if they are not already
+        # this enables use of only 1 set of parameters
+        (
+            names,
+            sensitivities,
+            custom_cost_segment_weight_factors,
+            allow_missing_osmids,
+        ) = self.convert_params_to_lists(
+            names,
+            sensitivities,
+            custom_cost_segment_weight_factors,
+            allow_missing_osmids,
+        )
+        self.validate_custom_cost_params(
+            names,
+            sensitivities,
+            custom_cost_segment_weight_factors,
+            allow_missing_osmids,
+        )
+        self.names = names
+        self.sensitivities = sensitivities
+        self.custom_cost_segment_weight_factors = custom_cost_segment_weight_factors
+        self.allow_missing_osmids = allow_missing_osmids
+        # GTFS is currently not supported for custom cost transport network
+        super().__init__(osm_pbf, gtfs=[])
+
+    def convert_params_to_lists(
+        self,
+        names,
+        sensitivities,
+        custom_cost_segment_weight_factors,
+        allow_missing_osmids,
+    ):
+        """
+        Convert single items into lists if they are not already.
+
+        Arguments:
+        ----------
+        names : str | List[str]
+            Single name or names of the custom costs.
+        sensitivities : float | int | List[float | int]
+            Single or multiple sensitivities of the custom costs.
+        custom_cost_segment_weight_factors : Dict[str, float] | List[Dict[str, float]]
+            Single or multiple custom cost factors to be used in routing.
+        allow_missing_osmids : bool | List[bool], default True (optional)
+            Define whether to allow missing osmids.
+
+        Returns:
+        --------
+        names : List[str]
+        sensitivities : List[float | int]
+        custom_cost_segment_weight_factors : List[Dict[str, float]]
+        allow_missing_osmids : List[bool]
+        """
+        if not isinstance(names, list):
```

Comment:
This does not work if, e.g., a tuple or another Iterable other than list is passed. Use `isinstance(maybe_a_list, collections.abc.Iterable)`

Reply:
I assume that it’s not necessary that the parameters are lists, specifically, but just behave like lists?=

Reply:
that's true and smart, i added Iterable support for params and some additional conditionals for e.g. string and dict (?) are iterable so they needed still checks not to pass faulty